### ISSUE TITLE
movies shorter than 1s give an error

### ISSUE THREAD

**Issue by dev-jam** [dschreij/python-mediadecoder#4]
When playing movies that are shorter than 1s I get an error about the max_duration in timer.py:

 max_duration needs to be greater than 1.0

The goal of this condition is unclear to me. Removing the condition fixes the problem (but maybe creates problems in other cases).

**Comment by dschreij** [comment]
It was intended to be a failsafe for when people try to load corrupted movies (they have no duration indication), or files that are no movie at all. I never expected anyone to play videos shorter than 1 second. If it doesn't break anything in your case, I will consider removing this check, although I should test a couple of edge-case scenarios first...

**Comment by dev-jam** [comment]
Maybe change the value to something really small like 0.0001?

**Comment by dschreij** [comment]
Yes, or maybe even 0.5s would suffice. I doubt people will ever want to play movies shorter than that (or maybe I'm falling into the same trap again).
Would it be possible for you to play around with this setting, test if it works correctly and submit a PR? Otherwise I'll see if I can work on this in the weekend.

**Comment by dschreij** [comment]
Still haven't had a chance to look at this, but it's not gone from my radar...

**Comment by fladd** [comment]
Maybe related to this: Seeking to 0.0 (and rewinding using the rewind method) will actually seek to 0.5 currently. It is not entirely clear to me why this is the case. There is a code comments stating "Make sure the movie starts at 1s as 0s gives trouble.". @dschreij Would you maybe be able to elaborate a bit on what that trouble is? I made some quick tests, but I didn't run into any issues so far. If I had some more insights on what the issues related to seeking to 0.0 are, I could have a closer look into this and see if this can be fixed (and along the way then also remove the 1 sec movie length restriction).
